fix(ui): hide "Ways to help" section under a spoiler

embarq opened the pull request:
Closes https://github.com/selfdefined/web-app/issues/325

embarq committed:
fix(ui): hide "Ways to help" section under a spoiler

embarq commented:
It was made to hide the "Ways to help" content only on mobile. I found that for the parent section `800px` breakpoint is used, so I took the same value into consideration.

Currently, the script will collapse the section on <800px screens, for desktop it will look as it was before.

However, the spoiler behaviour won't change on resize. For example if you open the homepage on mobile-like screen, and then resize it to let's say 810px - the spoiler will remain working(perhaps it should not). If you would like to have it adjusting to resize, let me know.

ovlb commented on 11ty/index.njk:
```diff
@@ -15,9 +15,24 @@ layout: layouts/index.njk
+    data-spoiler-mobile-only="true"
+    id="ways-to-help">
+    <h2 class="sub-headline">Ways to help
+      <button
+        aria-expanded="true"
+        aria-controls="ways-to-help-content"
+        class="spoiler__toggle-btn"
+        id="ways-to-help-visibility-toggle">
+        {# https://css.gg/chevron-down #}
+        <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" fill="none" viewBox="0 0 24 24"><path fill="currentColor" d="M6.34 7.76L4.93 9.17 12 16.24l7.07-7.07-1.41-1.41L12 13.4 6.34 7.76z"/></svg>
+      </button>
+    </h2>
```
Blocking: The button has no accessible name at the moment ([see this article for reference](https://www.sarasoueidan.com/blog/accessible-icon-buttons/)). We need to add some to have it make sense for screen reader users.

embarq replied:
The article was quite helpful, thanks a lot for sharing!

embarq replied:
Fixed in c73385a

ovlb commented on assets/css/layouts/_homepage.scss:
```diff
@@ -4,4 +4,16 @@
\ No newline at end of file
+}
+
+#ways-to-help.spoiler {
+  .sub-headline {
+    align-items: center;
+    display: flex;
+    justify-content: space-between;
+
+    .spoiler__toggle-btn {
+      color: #fff;
+    }
+  }
+}
```
I wonder if we can make these style more reusable, and less tied to the specific instance?

embarq replied:
Do you mean both `.sub-headline` and `.spoiler__toggle-btn`?

ovlb replied:
Yes, the combination of both. Aligned with the proposed update of the structure (https://github.com/selfdefined/web-app/pull/400#discussion_r621928910) you should be able to remove the nested styles altogether. Maybe add a utility class `.flex-space-between` to have it reusable and apply this to the header. Am I making sense? Have a bit of a brain freeze currently.

ovlb replied:
Another thing, for line 16. We added a dark mode last year, and as such, we try to avoid hardcoded colour values. The [links in the box are styled in this file](https://github.com/selfdefined/web-app/blob/prod/assets/css/components/_box.scss#L17), maybe it works to apply the same styles to the button?

embarq replied:
As for utility classes, here are related commits: 641120c, ff605f6
As for wrong icon color, I choose accent color. I think it looks even cooler 😁 . 8b69cce

ovlb commented on 11ty/index.njk:
```diff
@@ -15,9 +15,24 @@ layout: layouts/index.njk
+    <ol
+      class="list spoiler__content"
+      id="ways-to-help-content"
+      aria-labelledby="ways-to-help-visibility-toggle">
```
Blocking: This `aria-labelledby` does not work, as it’s pointing to the empty button. If we add toggle text to the button, this is probably not what we want as a label for the list. I’d propose to point to the headline, and move the button out of the headline (e.g. having the button and headline in a `header`. 

```suggestion
      aria-labelledby="ways-to-help"
```

embarq replied:
Fixed in c73385a

ovlb reviewed (changes requested):
Hey! Thanks so much. This will come in handy. I’ve left some accessibility-related notes. :)

embarq committed:
fix(styles): add flex utilities classes

embarq committed:
fix(styles): remove unecessary styles

To achieve the same goal, removed
logic is moved to `assets/css/base/_helpers.scss`
in form of flexbox utitlity classes.
Later, these utility classes will be used for the homepage layout.

Utility classes were introduced in
`641120ce1f4060ceb4c20be7c560969da432055e`

embarq committed:
fix(styles): use app's accent color for spoiler toggle

embarq committed:
fix(a11y): "ways to help" structure

- use correct a11y attributes and values
- fix code formatting
- use HTML structural elements to make markup more descriptive

embarq commented:
@ovlb Man, I can't express how grateful I am. Thanks for so many useful and interesting suggestions!
I tried my best in attempt to make the implementation better, would be awesome if you could check the result once again.

P.S. Thanks for being so helpful 👍 As you might notice, I have quite a limited knowledge of a11y and how to implement it in a right way. Looking forward to learn and practise more tho

ovlb committed:
Merge branch 'prod' into fix/home-ways-to-help

ovlb commented:
@embarq Thanks for your updates. I had quite a packed week, so please bear with me. I want to review it with the time it deserves. Aiming for Monday. Hope you have a nice weekend. 

O*
